skimage/transform/registration.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+from scipy.optimize import minimize
+from scipy import ndimage as ndi
+
+from .pyramids import pyramid_gaussian
+from ..measure import compare_nmi
+
+__all__ = ['register_affine']
+
+
+def _parameter_vector_to_matrix(parameter_vector, N):
+    """
+    Converts the optimisation parameters to a 3x3 transformation matrix
+
+    The optimisation paramters are known as the parameter_vector and are
+    composed of the first 2 rows of the transformation matrix, as that is
+    all that is used in an affine transformation.
+
+    Parameters
+    ----------
+    parameter_vector : (N*(N+1)) array
+        Input array giving the argument of the minimum function to
+        optimise against
+
+    Returns
+    -------
+    matrix : (N+1, N+1) array
+        A transformation matrix used to obtain a new image
+    """
+
+    return np.concatenate(
+        (np.reshape(parameter_vector, (N, N + 1)), [[0] * N + [1]]), axis=0)
+
+
+def _matrix_to_parameter_vector(matrix):
+    """
+    Converts a (N+1)x(N+1) transformation matrix to the optimisation parameters
+
+    See the inverse function `_parameter_vector_to_matrix`.
+
+    Parameters
+    ----------
+    matrix : (N+1, N+1) array
+        A transformation matrix used to obtain a new image
+
+    Returns
+    -------
+    parameter_vector : (N*(N+1)) array
+        Output array giving the argument of the minimum function to
+        optimise against
+
+    """
+
+    return matrix[:-1, :].ravel()
+
+
+def cost_nmi(image0, image1, *, bins=100):
+    """Negative of the normalized mutual information.
+
+    See :func:`skimage.measure.compare_nmi` for more info.
+
+    Parameters
+    ----------
+    image0, image1 : array
+        The images to be compared. They should have the same shape.
```

Comment:
I would support also differently sized images and add padding to smaller one or give user such function... it is a very common case that the images are not the same size...

Reply:
👍